interfaces/machine_parameters/ring_section.py

```diff
@@ -177,98 +165,134 @@ def __init__(self, section_length, alpha_0,
 
         # Reshaping the input synchronous data to the adequate format and
         # get back the momentum program
-        if not isinstance(synchronous_data, dTypes._ring_program):
+        if not isinstance(synchronous_data,
+                          ring_programs._synchronous_data_program):
             synchronous_data \
-                = dTypes._ring_program.conversions[func_type](synchronous_data)
+                = ring_programs._synchronous_data_program._conversions[
+                    func_type](synchronous_data)
 
         self.synchronous_data = synchronous_data
 
+        # Setting orbit length
+        if orbit_bump is None:
+            self.length = ring_programs.orbit_length_program(
+                self.length_design)
+        else:
+            if not isinstance(orbit_bump, ring_programs.orbit_length_program):
+                orbit_bump = ring_programs.orbit_length_program(orbit_bump)
+            if orbit_bump.timebase == 'by_time':
+                orbit_bump[:, 1, :] += self.length_design
+            else:
+                orbit_bump += self.length_design
+            self._check_and_set_alpha_and_orbit(orbit_bump)
+
         # Setting the linear momentum compaction factor
         # Checking that the synchronous data and the momentum compaction
         # have the same length if defined turn-by-turn, raise a warning if one
         # is defined by turn and the other time based
-        self.alpha_order = 0
-        if not isinstance(alpha_0, dTypes.momentum_compaction):
-            alpha_0 = dTypes.momentum_compaction(alpha_0, order=0)
+        self.alpha_orders = [0]
+        alpha_order_max = 0
+        if not isinstance(alpha_0, ring_programs.momentum_compaction):
+            alpha_0 = ring_programs.momentum_compaction(alpha_0, order=0)
         else:
             if alpha_0.order != 0:
                 raise excpt.InputError(
                     "The order of the datatype passed as keyword " +
                     "argument alpha_%s do not match" % (0))
 
-        self._check_and_set_momentum_compaction(alpha_0, 0)
+        self._check_and_set_alpha_and_orbit(alpha_0, 0)
 
         # Treating non-linear momentum compaction factor if declared
         # Listing all the declared alpha first
         alpha_n = {1: alpha_1, 2: alpha_2}
 
         if alpha_1 is not None:
-            self.alpha_order = 1
+            alpha_order_max = 1
         if alpha_2 is not None:
-            self.alpha_order = 2
+            alpha_order_max = 2
 
         for argument in kwargs:
             if 'alpha' in argument:
                 try:
                     order = int(argument.split('_')[-1])
-                    self.alpha_order = np.max([self.alpha_order, order])
+                    alpha_order_max = np.max([alpha_order_max, order])
                     alpha_n[order] = kwargs[argument]
                 except Exception:
```

Comment:
Eeeesh.  `except Exception` is risky...

Reply:
Replaced Exception by ValueError (the goal here was to catch wrong casting to`int(argument.split('_')[-1])`, all cases I can think of lead to ValueError but I set to Exception originally for cases I wouldn't have considered)

Reply:
You're right, I tested with impossible input, since they're always coming from kwargs it'll never be anything but `str`.  If a `TypeError` ever comes up there's a MUCH bigger problem than a bad kwarg.